src/vector/ops/set.rs

```diff
@@ -33,19 +40,25 @@ impl Geometry {
         Some(unsafe { Geometry::with_c_geometry(ogr_geom, true) })
     }
 
-    /// Compute union.
+    /// Computes the [geometric union][union] of `self` and `other`.
     ///
-    /// Generates a new geometry which is the region of union of
-    /// the two geometries operated on.
-    /// Geometry validity is not checked. In case you are unsure of the
-    /// validity of the input geometries, call IsValid() before,
-    /// otherwise the result might be wrong.
+    /// Generates a new geometry which is the union of the two geometries operated on.
+    ///
+    /// # Notes
+    /// * Geometry validity is not checked, and invalid geometry will generate unpredictable results.
+    /// Use [`Geometry::is_valid`] if validity might be in question.
+    /// * If GEOS is *not* enabled, this function will always return `None`.
+    /// You  may check for GEOS support with [`VersionInfo::has_geos`][has_geos].
     ///
     /// # Returns
-    /// `Some(geometry)` if both Geometries contain pointers.
-    /// `None` if either geometry is missing the GDAL pointer, or there is an error.
+    /// * `Some(geometry)`: a new `Geometry` representing the computed union
+    /// * `None`: when the union was not able to be computed
```

Comment:
I'm really frustrated by this, as the C/C++ docs say: 

> a new geometry representing the union or NULL if an error occurs.

which would imply to me that we should be returning `Result<Geometry, GDALError>`.

Reply:
GDAL sets an error message when GEOS or SFCGAL are not available, but then the "correct" return type would be `Result<Option<Geometry>, GDALError>`, right? Which is a little annoying.

It might be a moot point. I suspect it's impossible to get a GEOS-less GDAL unless you're compiling it yourself, but then not everyone might know that GEOS is important to have when you're using vector data.

Also, I wonder if we should mention SFCGAL, which is required for [some geometry types](https://github.com/OSGeo/gdal/blob/e708608cf2466580db2690fa93cb3823ded6d4a3/ogr/ogrgeometry.cpp#L7469).

Reply:
🤔